lib/internal/http/static.js

```diff
@@ -0,0 +1,144 @@
+'use strict';
+
+const {
+  Promise,
+  RegExpPrototypeExec,
+  StringPrototypeEndsWith,
+  StringPrototypeIncludes,
+  StringPrototypeIndexOf,
+  StringPrototypeLastIndexOf,
+  StringPrototypeReplace,
+  StringPrototypeSlice,
+  StringPrototypeStartsWith,
+  StringPrototypeToLocaleLowerCase,
+} = primordials;
+
+const { cwd } = process;
+const console = require('internal/console/global');
+const { createReadStream } = require('fs');
+const { Server } = require('_http_server');
+const { URL, isURLInstance, pathToFileURL } = require('internal/url');
+const { emitExperimentalWarning, kEmptyObject } = require('internal/util');
+const { validateFunction, validatePort } = require('internal/validators');
+
+function filterDotFiles(url, fileURL) {
+  return !StringPrototypeIncludes(fileURL.pathname, '/.');
+}
+
+function createStaticServer(options = kEmptyObject) {
+  emitExperimentalWarning('http/static');
+
+  const {
+    directory = cwd(),
+    port,
+    host = 'localhost',
+    mimeOverrides,
+    filter = filterDotFiles,
+  } = options;
+  const directoryURL = isURLInstance(directory) ? directory : pathToFileURL(directory);
+  // To be used as a base URL, it is necessary that the URL ends with a slash:
+  const baseDirectoryURL = StringPrototypeEndsWith(directoryURL.pathname, '/') ?
+    directoryURL : new URL(`${directoryURL}/`);
+
+  if (port != null) validatePort(port);
+  if (filter != null) validateFunction(filter, 'options.filter');
+
+  const server = new Server(async (req, res) => {
+    const url = new URL(
+      req.url === '/' || StringPrototypeStartsWith(req.url, '/?') ?
+        './index.html' :
+        StringPrototypeReplace(new URL(req.url, 'root://').toString(), 'root://', '.'),
+      baseDirectoryURL
+    );
+
+    if (filter != null && !filter(req.url, url)) {
+      console.log('403', req.url);
```

Comment:
If we are exposing this as a public function, we should not be using `console.log()` in it. Logging should be configurable somehow.